irc/server.go

```diff
@@ -1453,6 +1318,18 @@ func (server *Server) rehash() error {
 		}
 	}
 
+	// sanity checks complete, start modifying server state
+
+	server.name = config.Server.Name
+	server.nameCasefolded = casefoldedName
+	server.networkName = config.Network.Name
```

Comment:
We should make sure that server name can't change after launching the server, and error out with something like the maximum linelen message above. I'd be okay with the network name changing, so long as a new `NETWORK=blah` RPL_ISUPPORT token is sent (should be but good to confirm).